### PR TITLE
Support SQS as a trigger for functions

### PR DESCRIPTION
**Pull request by SiniWilson**
### Requirements

Support SQS as a trigger for functions

### Benefits

User should be able to select SQS as function trigger.

### CODE REVIEW

File: builds/jenkins-build-pack-lambda/Jenkinsfile [hunk 1]
```diff
@@ -330,6 +313,50 @@ node() {
...
+    error "Exception occured while serverless deployment to ${environment_logical_id} environment"
+  }
+}
+
+def handleSqsError(deployOutput, event_source_sqs_arn, environment_logical_id, envBucketKey) {
+  def sqsErrString = "CloudFormation - CREATE_FAILED - AWS::SQS::Queue - "
+  if (deployOutput.contains(sqsErrString)) {
+    def event_source_sqs = event_source_sqs_arn.split(":(?!.*:.*)")[1]
+    if (lambdaEvents.checkSqsQueueExists(event_source_sqs)) {
+      lambdaEvents.updateSqsResourceServerless()
```
Comment (svidhani) [inline builds/jenkins-build-pack-lambda/Jenkinsfile]: This code should work when sqs queue exists. So why are you adding the queue here. i am confused

File: builds/jenkins-build-pack-lambda/Jenkinsfile [hunk 2]
```diff
@@ -330,6 +313,50 @@ node() {
...
+      echoServerlessFile()
+      sh "serverless remove --stage $environment_logical_id -v --bucket ${configLoader.AWS.S3[envBucketKey]}"
+      def redeployOutput = servelessDeploy(environment_logical_id, envBucketKey)
+      if (redeployOutput != 'success') {
+        echo "Exception occured while serverless deployment to ${environment_logical_id} environment : $redeployOutput"
+        error "Exception occured while serverless deployment to ${environment_logical_id} environment"
+      }
+    } else {
+      error "Error occured while accessing the kinesis stream"
```
Comment (svidhani) [inline builds/jenkins-build-pack-lambda/Jenkinsfile]: why kinesis error?

Comment (svidhani) [reply builds/jenkins-build-pack-lambda/Jenkinsfile]: also if the queue does not exist shouldn't you be creating a new queue here?

File: builds/jenkins-build-pack-lambda/Jenkinsfile [hunk 3]
```diff
@@ -554,8 +583,8 @@ def loadServerlessConfig(String runtime, def isEventSchdld, def isScheduleEnable
     sh "cp _config/serverless-python.yml ./serverless.yml"
   }
 
-  if (isDynamoDbEnabled) {
-    sh "cp _config/aws-events-policies/dynamodb-policy.yml ./policyFile.yml"
+  if (isDynamoDbEnabled || isSQSEventEnabled) {
+    sh "cp _config/aws-events-policies/custom-policy.yml ./policyFile.yml"
```
[the remark this answers is not in the document]
Comment (SiniWilson) [reply builds/jenkins-build-pack-lambda/Jenkinsfile]: The custom role is not needed for kinesis

File: builds/jenkins-build-pack-lambda/Jenkinsfile (hunk 2 above)
Comment (SiniWilson) [reply builds/jenkins-build-pack-lambda/Jenkinsfile]: corrected the error message. If the queue not exists, serverless will create new queue by the resources option in the serverless

File: builds/jenkins-build-pack-lambda/Jenkinsfile [hunk 4]
```diff
@@ -330,6 +313,50 @@ node() {
...
 
+def handleS3BucketError(deployOutput, s3BucketName, environment_logical_id, envBucketKey, isEventSchdld) {
+  def s3BucketErrString = "CloudFormation - CREATE_FAILED - AWS::S3::Bucket - "
+  if (deployOutput.contains(s3BucketErrString)) {
+    if (lambdaEvents.checkS3BucketExists(s3BucketName)) {
+      isLambdaUpdateRequired = true
+      lambdaEvents.removeS3EventsFromServerless(isEventSchdld)
+      echoServerlessFile()
+      sh "serverless remove --stage $environment_logical_id -v --bucket ${configLoader.AWS.S3[envBucketKey]}"
+      def redeployOutput = servelessDeploy(environment_logical_id, envBucketKey)
```
[the remark this answers is not in the document]
Comment (SiniWilson) [reply builds/jenkins-build-pack-lambda/Jenkinsfile]: done. Will raise a follow up PR for kinesis - I am expecting merge conflict for these PRs.

**Comment by svidhani** [comment]
It seems DynamoDB is broken probably due to your changes. Please syncup with @supritat to ensure its working in the stack.

**Comment by SiniWilson** [comment]
@svidhani @supritat Please share the stack details and build numbers for the broken dynamodb, So that I can fix the issue and raise a follow up PR. I think it will be better to use slack channel for conveying these type of messages rather than CR comments.

**Pull request merged**